### PR TITLE
fix(api/notification): correction of contact and contact group inheritance

### PR DESCRIPTION
**Pull request by callapa**
## Description

When calling API `configuration/hosts/{host_id}/services/{service_id}/notification-policy`, the inheritance calculation for contacts and contact groups was not correct when the calculation mode was based on "Cumulative inheritance".

**Fixes** # MON-32888

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [x] 23.04.x
- [x] 23.10.x
- [x] 24.04.x (master)

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Comment by cgagnaire** [comment]
Hi @callapa,
Does it fix what is displayed on Notification tab ?
Thanks!

File: centreon/src/Centreon/Infrastructure/MonitoringServer/Repository/MonitoringServerConfigurationRepositoryApi.php [hunk 1]
```diff
@@ -163,9 +157,9 @@ private function callHttp(string $filePath, string $payloadBody): void
                 'proxy' => null,
                 'no_proxy' => '*',
                 'verify_peer' => false,
-                'headers' => ['X-AUTH-TOKEN' => $providerToken->getToken()],
+                'headers' => ['X-AUTH-TOKEN' => $providerToken->getToken(), 'Cookie' => 'XDEBUG_SESSION=XDEBUG_KEY'],
```
Comment (arnapou) [inline centreon/src/Centreon/Infrastructure/MonitoringServer/Repository/MonitoringServerConfigurationRepositoryApi.php:160]: XDEBUG !?

**Comment by callapa** [comment]
Hi @cgagnaire,
> Does it fix what is displayed on Notification tab ?

Yes

**Review by arnapou — APPROVED** [review]

**Review by arnapou — APPROVED** [review]

**Review by a-launois — APPROVED** [review]

**Review by dmyios — APPROVED** [review]

**Comment by cgagnaire** [comment]
Re-hi @callapa,
Moreover, does it fix the cumulative inheritance from both host/host's templates and service's templates for a service when configuration is generated ?
Thanks!

**Pull request merged**